src/Config.ts

```diff
@@ -102,19 +74,126 @@ export class Chain {
   public rpc: string;
 
   /** Worker account object. */
-  public worker: Account;
+  public worker: string;
+}
+
+/**
+ * It holds contents of the facilitator config.
+ */
+export class FacilitatorConfig {
+  public database: DBConfig;
+
+  public chains: Record<string, Chain>;
+
+  public encryptedAccounts: Record<string, any>;
+
+  private static fileName: string = 'facilitator-config.json';
 
   /**
    * Constructor.
-   * @param chainId Chain id.
-   * @param facilitatorConfig Facilitator config json data.
+   * @param config Facilitator config object.
+   */
+  private constructor(config: any) {
+    this.database = config.database || new DBConfig();
+    this.chains = config.chains || {};
+    this.encryptedAccounts = config.encryptedAccounts || {};
+  }
+
+  /**
+   * It writes facilitator config object.
+   * @param {string} chain Auxiliary chain id.
+   */
+  public writeToFacilitatorConfig(chain: string): void {
+    const mosaicConfigDir = Directory.getMosaicDirectoryPath();
+    const configPath = path.join(
+      mosaicConfigDir,
+      chain,
+    );
+    fs.ensureDirSync(configPath);
+
+    fs.writeFileSync(
+      path.join(configPath, FacilitatorConfig.fileName),
+      JSON.stringify(this, null, '    '),
+    );
+  }
+
+  /**
+   * This reads mosaic config from the json file and creates FacilitatorConfig object.
+   * @param {string} chain Auxiliary chain id.
+   * @returns {FacilitatorConfig} Facilitator config object.
    */
-  public constructor(readonly chainId: string, facilitatorConfig: Record<string, any>) {
-    const chainData = facilitatorConfig.chains[chainId];
-    this.rpc = chainData.rpc;
-    this.worker = new Account(
-      chainData.worker,
-      facilitatorConfig.encrypted_accounts[chainData.worker],
+  public static from(chain: string): FacilitatorConfig {
+    const facilitatorConfigPath = path.join(
+      Directory.getMosaicDirectoryPath(),
+      chain,
+      FacilitatorConfig.fileName,
     );
+
+    if (fs.existsSync(facilitatorConfigPath)) {
+      const config = fs.readFileSync(facilitatorConfigPath).toString();
+      if (config && config.length > 0) {
+        const jsonObject = JSON.parse(config);
+        return new FacilitatorConfig(jsonObject);
+      }
+    }
+    return new FacilitatorConfig({});
+  }
+
+  /**
+   * @returns {FacilitatorConfig} FacilitatorConfig object.
+   */
+  public static new(): FacilitatorConfig {
+    return new FacilitatorConfig({});
+  }
+
+  /**
+   * It verifies if facilitator config is present for a auxiliary chainid. If already present then it exits.
+   * @param {string} chain Auxiliary chain id.
+   */
+  public static assertNotExists(chain: string): void {
```

Comment:
This function should not be an assert function, it can be the function that can check if the facilitator config exists for a given chain id. 

Else move this function from this file to facilitator-init.

I think that this should just return boolean if exists or not.
The caller can decide what to do.
The logs indicate that it is very specific to facilitator init.

Reply:
Changed the name to `isFacilitatorConfigPresent`.